### DISCUSSION TITLE
Enable Merge Strategy per Branch

### DISCUSSION THREAD

**Discussion by AllanOricil** [community/community#53985] [Repositories]
### Select Topic Area

Product Feedback

### Body

In certain git branching models, like git flow, we can choose different merge strategies depending on the "branch" we are merging to. Unfortunatelly Github does not let us map the merge  strategy by branch. I would like to have something like the following as part of the "Branch Protection Rules"

- PRs to `main` can only be merged with "rebase and merge"
- PRs to `release/*` can only be merged with "squash and merge"
- PRs to `develop` can only be merged with "squash and merge"

The PR merge button strategy is currently a repository settings and it does not allow configuration per branch
<img width="749" alt="image" src="https://user-images.githubusercontent.com/55927613/235203352-ef204deb-93c6-48ea-ae4c-12c87856cb66.png">

**Comment by iamtodor** [comment]
Related to:
- https://github.com/orgs/community/discussions/20583
- https://github.com/orgs/community/discussions/10809
- https://github.com/orgs/community/discussions/20583
